src/main/resources/include/hail/NativePtr.h

```diff
@@ -0,0 +1,78 @@
+#ifndef HAIL_NATIVEPTR_H
+#define HAIL_NATIVEPTR_H 1
+
+#include "hail/NativeObj.h"
+#include <jni.h>
+#include <memory>
+
+namespace hail {
+
+typedef long LongFuncN(...);
+typedef NativeObjPtr PtrFuncN(...);
+
+template<typename ReturnT>
+class NativeFuncObj : public NativeObj {
+public:
+  typedef ReturnT FuncType(...);
+public:
+  NativeObjPtr module_; // keep-alive for the loaded module
+  FuncType *func_;
+  
+public:
+  inline NativeFuncObj(
+    NativeObjPtr module,
+    void* funcAddr
+  ) :
+    module_(module) {
+    // It's awkward to turn a "void*" into a function pointer
+    void** pFunc = reinterpret_cast<void**>(&func_);
+    *pFunc = funcAddr;
+  }
+  
+  virtual ~NativeFuncObj() { }
+
+private:
+  // disable copy-assign
+  NativeFuncObj& operator=(const NativeFuncObj& b);
```

Comment:
I think the preferred C++11 way to disable compiler-generated methods is
```C++
public:
  NativeFuncObj& operator=(const NativeFuncObj& b) = delete;
```
See e.g. https://abseil.io/tips/143

Reply:
OK.  That's some new-fangled C++11 right there, and we old-timers have been doing it the way our grandmothers learned us back in the day.

Reply:
Another discussion is Item 11 in Meyers' "Effective Modern C++", titled "Prefer deleted functions to private undefined ones." (This is a great book, I highly recommend glancing through it once in a while. If it's on my desk, you're always welcome to grab it. It belongs to the office.)

The biggest reason for adding it to the language is that deleted functions apply in situations that pre-C++11 had no solution for. But even in this case, delete has advantages, e.g. even code in member and friend functions will fail to compile if it tries to copy a `NativeFuncObj`.